### PR TITLE
Option to abort request when kv cache miss

### PR DESCRIPTION
**Pull request by jerrychenhf**
## Purpose
The current DP implementation cannot handle the case of not bypass prefill model forward when KV cache miss.
This allows to abort the request when async KV cache fetching failure for the request, instead of adding the request to normal execution.

### CODE REVIEW

File: vllm/core/scheduler.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 import torch
 
+import vllm.envs as envs
```
[the remark this answers is not in the document]
Comment (jerrychenhf) [reply vllm/core/scheduler.py:17]: deepseek_r1 branch is an old branch with code not compatible with current formatting rules.
We make sure the new code follow the formatting rules while will not fix the old ones.
